### DISCUSSION TITLE
How to read environment variables in a Node project?

### DISCUSSION THREAD

**Discussion by imdavidmin** [deta/cloud-docs#481] [Help]
In my JS code I would like to read from an environment variable I've set on the deta CLI. 

With `import { os } from 'os';` , `os.getenv` is showing as not valid with typescript checking. `process.env` wasn't working before.

**Comment by Sokhavuth** [comment]
You need to install dotenv module first
```js
// npm install dotenv

const dotenv = require('dotenv')
dotenv.config()

// using environment variable SECRET_KEY
const secret_key = process.env.SECRET_KEY
```

**Comment by imdavidmin** [comment]
Please delete this thread - `process.env` works without using `dotenv`. My error was generated from somewhere else.

**Reply by Sokhavuth** [reply]
You do not use .env file?
[https://www.npmjs.com/package/dotenv](https://www.npmjs.com/package/dotenv)
